### ISSUE TITLE
`az storage blob sync` requires root privileges

### ISSUE THREAD

**Issue by jlorich** [Azure/azure-cli#10024]
## Describe the bug

The first run of `az storage blob sync` attempts to place the `azcopy` executable in `/usr/local/bin`, which is only accessible with elevated privileges.  This is not ideal as root is not always available.  If there's a dependency on `azcopy` it should probably be installed along with the CLI.

`sudo` under the `AzureCLI@1` Azure DevOps task also appears to break authentication, so it's not a viable option in CI:

```
sudo az storage blob sync -s /home/vsts/work/1/blog -c '$web' --account-name jlorichblogst |  
ERROR: Please run 'az login' to setup account.
```

**Command Name**
`az storage blob sync`

**Errors:**
```
Connection error while attempting to download azcopy ([Errno 13] Permission denied: '/usr/local/bin/azcopy')
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Run the following as a non-privileged user who has not used anything requiring azcopy before
- `az storage blob sync -s {} -c {} --account-name {}`

## Expected Behavior
Sync files with blob storage without needing additional permissions.

## Environment Summary
```
Linux-4.4.0-18362-Microsoft-x86_64-with-debian-buster-sid
Python 3.6.5
Shell: bash

azure-cli 2.0.68 *

```

**Comment by jlorich** [comment]
Even with an additional installation step for `azcopy` in CI this still fails.  It only succeeds when running the installation and then creating a symbolic link to the `/usr/local/bin/azcopy` path the cli is looking for.
```
    - script: |
        wget -O azcopy.tar.gz https://aka.ms/downloadazcopylinux64
        tar -xf azcopy.tar.gz
        sudo ./install.sh
        sudo ln -s /usr/bin/azcopy /usr/local/bin/azcopy
```

With that step placed ahead, my Azure CLI task now works as intended:

```
    - task: AzureCLI@1
      inputs:
        azureSubscription: 'Visual Studio Enterprise(e97d8b6e-d05b-4507-9c06-f7b528f65f7d)'
        scriptLocation: 'inlineScript'
        inlineScript: |
          # Fail on errors, output every command and result to the log
          set -e -x

          # Configure static sites
          az storage blob service-properties update --account-name $(storageAccountName) --static-website  --index-document index.html --404-document 404.html

          # Upload files
          az storage blob sync \
            -s $(Pipeline.Workspace)/blog \
            -c \$web \
            --account-name $(storageAccountName)
```

I don't consider this a real solution, as the Azure Cli should have a more robust answer for finding and installing `azcopy`, but it this serves as a sufficient workaround.

**Comment by Juliehzl** [comment]
@limingu can you please take a look?

**Comment by veleek** [comment]
@jlorich - I was unable to get your workaround working due to issues setting up the symbolic link to azcopy.  Is there a chance that azcopy can already exist on the machine?  If so, should that existing copy be used by the azcli task, or will it still try to copy itself over?

Is there a way to force it to use azcopy in the local directory instead of the global one?

**Comment by Juliehzl** [comment]
Hi @jlorich @veleek, in azure cli 2.0.71 which will be released on this Tuesday, we change the installation location of azcopy to local directory, so the root privilege will not be needed.

**Comment by Juliehzl** [comment]
@jlorich Already fixed in zure cli 2.0.71.

**Comment by veleek** [comment]
Was finally able to confirm that this is all working properly now.  Cut 2 minutes off of my build times by being able to switch to Linux machines! :)
